File: src/main/kotlin/com/screentech/cordamigrate/entity/securityquestion/SecurityQuestion.java
package com.screentech.cordamigrate.entity.securityquestion;

import java.math.BigDecimal;
import java.sql.Date;
import javax.persistence.*;

@Entity
@Table(name = "security_questions")
public class SecurityQuestion{

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) @Column(name = "id")
    private BigDecimal id;

    @Column(name = "question")
    private String question;

    @Column(name = "the_timestamp", updatable = true, insertable = true)
    private Date timestamp;

    public SecurityQuestion(){

    }

    public BigDecimal getId() {
        return id;
    }

    public void setId(BigDecimal id) {
        this.id = id;
    }

    public String getQuestion() {
        return question;
    }

    public void setQuestion(String question) {
        this.question = question;
    }

    public Date getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Date timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return "SecurityQuestion{" +
                "id=" + id +
                ", question='" + question + '\'' +
                ", timestamp=" + timestamp +
                '}';
    }
}

File: src/main/kotlin/com/screentech/cordamigrate/entity/securityquestion/SecurityQuestion.kt
package com.screentech.cordamigrate.entity.securityquestion

import java.sql.Date
import java.sql.Timestamp
import javax.persistence.*

@Entity
@Table(name = "security_questions")
class SecurityQuestion(
        @Id @GeneratedValue(strategy = GenerationType.IDENTITY) @Column(name = "id") var id: Long?,
        @Column(name = "question", columnDefinition = "text") var question: String?,
        @Column(name = "the_timestamp", columnDefinition = "timestamp default now()") var timestamp: Timestamp?,
        @Transient var timestampStr : String?
) {

    override fun toString(): String {
        return "SecurityQuestion(id=$id, question=$question, timestamp=$timestamp, timestampStr=$timestampStr)"
    }
}
